exec/java-exec/src/main/java/org/apache/drill/exec/rpc/user/UserSession.java

```diff
@@ -97,6 +101,14 @@ public Builder withUserProperties(UserProperties properties) {
       return this;
     }
 
+    public Builder setSessionContextInfo(Map<String, String> sessionParams) {
+        UserSessionContextInformation.Builder sessionContextBuilder = UserSessionContextInformation.newBuilder()
+            .setSessionId(userSession.sessionId);
+        // TODO: Populate sessionParams into UserSesessionContextInfo
+        userSession.sessionContextInfo =  sessionContextBuilder.build();
+        return this;
+    }
```

Comment:
I guess UserSessionContextInformation should exist disregarding calling setSessionContextInfo. setSessionContextInfo needs to enrich it.